### DISCUSSION TITLE
Why is media not stored in the database

### DISCUSSION THREAD

**Discussion by ronsplinter** [PiranhaCMS/piranha.core#1418] [Ideas]
Is there a reason why the media is not stored in the database? I know that I can create a new implementation of the IStorage interface to store the data somewhere else but I would like to store it in the same database. I would like to built it as a contribution to the project. Are you open for this?

**Comment by tidyui** [comment]
In the core project we have decided **not** to store media in the database as it is an easy performance killer. Databases are optimized for storing relational data (at least the databases we provide packages for) and they are just not as good at storing big binary files. With this in mind we provide two options, either store media together with the web application and let it be handled by the `StaticFiles` middleware, or store it in `Azure BlobStorage` like we do in all of our production systems.
